src/main/java/ch/njol/skript/expressions/ExprSpawnEggEntity.java

```diff
@@ -0,0 +1,107 @@
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.aliases.ItemType;
+import ch.njol.skript.bukkitutil.ItemUtils;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.*;
+import ch.njol.skript.entity.EntityData;
+import ch.njol.skript.expressions.base.SimplePropertyExpression;
+import ch.njol.skript.util.slot.Slot;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.EntitySnapshot;
+import org.bukkit.event.Event;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.SpawnEggMeta;
+import org.jetbrains.annotations.Nullable;
+
+@Name("Spawn Egg Entity")
+@Description({
+	"Gets or sets the entity snapshot that the provided spawn eggs will spawn when used.",
+	"NOTE: When using an entity data (i.e 'a zombie'), it will spawn a new zombie to retrieve an entity snapshot and then be removed."
```

Comment:
```suggestion
	"NOTE: When using an entity data (i.e 'a zombie'), it will spawn a new zombie to retrieve an entity snapshot and then be removed."
```
should this even be mentioned? this feels like an internal detail

Reply:
Well, considering before the `EntityData#create` that I added, it was spawning the entity, so having a note saying that it would spawn the entity would be information for users in case they noticed an entity spawning and immediately despawning. But it can be removed now